### DISCUSSION TITLE
How to run the hindcast experiments using GCAM?

### DISCUSSION THREAD

**Discussion by tfukuda-ntt** [JGCRI/gcam-core#335] [Q&A]
I’m now designing an experiment in which the available amount of renewable freshwater (maxSubResource), currently set at a constant value for all years, varies every five years. 

As an example, I assume to change `maxSubResource = 38.80101` defined in `Resource=waterwithdrawals, sub-renewable-resource=runoff` in `water_supply_constraint.xml` as shown in the attached image from year to year.

![image-20230823-055444](https://github.com/JGCRI/gcam-core/assets/29192685/b4dce044-c72e-409f-9860-313502676c8d)

I want to perform a historical experiment (hindcast experiment) to compare the calculation results obtained from this change with real (actual) data. Currently, I would like to do it on data from 2010, 2015, and 2020, using 2005 as the final calibration year.

However, I’m having trouble figuring out how to run the hindcast experiments using GCAM. Could you please tell me how to do it? I’m using GCAM v.5.4, v.6.0 and v7.0.

**Comment by rubcalvo** [comment]
I was testing some GCAM runs today and I edited the `modeltime.xml` file to set the final calibration date to 2005 and the GCAM log says 

"Read in final-calibration-year (2005) earlier than last historical year (2015).
Running in HINDCASTING MODE with (2005) as the final calibration year." 

But it fails to solve 

Period 2: 2005
ERROR:Base year demand: 0.47709 is greater than satiation level: 0.462917

I don't know if this helps you with your question but I guess this is one way to force a hindcast mode in GCAM.

Best regards,

**Comment by pkyle** [comment]
Just changing `modeltime.xml` isn't something that should be expected to work. This specific crash/error is related to the buildings sector calibration of service demand functions, but even if you make a workaround that avoids that error, there will probably be other errors and/or logical inconsistencies. For example the nonCO2 emissions of most technologies will be prescribed for the 2010 and 2015 time periods, irrespective of input or output. Yields will probably be constant between 2005 and 2015 as `AgProdChange` defaults to 0.
The way to run in hindcast mode is to re-set the objects `MODEL_BASE_YEARS`, `MODEL_FINAL_BASE_YEAR`, and `MODEL_FUTURE_YEARS` in `gcamdata/R/constants.R` and re-build all XML files.
